### PR TITLE
Fixes #1040

### LINKED ISSUE #1040: Can't correctly show return-value type
- What version did you use?
v6.0.0.0
- Write down all steps needed to reproduce this issue. Also attach any needed files.
1. Open mscorlib.dll (.NET 4.0)
2. Jump to "System.SZArrayHelper"
3. Hover above "JitHelpers.UnsafeCast<T[]>(this);"

snipaste_2019-01-05_21-02-13

### PR DESCRIPTION
**Pull request by xsilent007**
This should fix the return types not being shown correctly according to the comment here: https://github.com/0xd4d/dnSpy/issues/1040#issuecomment-458599377

Closes #1040

### CODE REVIEW

**Comment by xsilent007** [comment]
Screenshot

**Comment by 0xd4d** [comment]
Thanks. The VB formatter needs the same fix too: https://github.com/0xd4d/dnSpy/blob/master/dnSpy/dnSpy.Decompiler/VisualBasic/VisualBasicFormatter.cs#L884

File: dnSpy/dnSpy.Decompiler/CSharp/CSharpFormatter.cs [hunk 1]
```diff
@@ -884,7 +884,7 @@ public struct CSharpFormatter {
 					type = type.Next;
 				}
 				if (list != null) {
-					Write(list[list.Count - 1].Next, typeGenArgs, methGenArgs);
+					Write(list[list.Count - 1].Next, typeGenArgs, new List<TypeSig>());
```
Comment (0xd4d) [inline dnSpy/dnSpy.Decompiler/CSharp/CSharpFormatter.cs]: nit: you can use `Array.Empty<TypeSig>()` so it won't allocate any memory.

**Comment by xsilent007** [comment]
Yup, I was dumb the first time.

**Comment by 0xd4d** [comment]
Thanks, it's been merged!

**Pull request merged**